uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/microsoftsts/TlsPromptHandler.java

```diff
@@ -0,0 +1,115 @@
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.microsoftsts;
+
+import android.text.TextUtils;
+import android.widget.ImageButton;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+
+import com.microsoft.identity.client.ui.automation.browser.BrowserChrome;
+import com.microsoft.identity.client.ui.automation.interaction.AbstractPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.logging.Logger;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.FIND_UI_ELEMENT_TIMEOUT;
+
+/**
+ * A Prompt Handler for TLS login flows.
+ */
+public class TlsPromptHandler extends AbstractPromptHandler {
+
+    private final static String TAG = TlsPromptHandler.class.getSimpleName();
+    public final static long CHROME_MENU_BUTTON_RETRY_TIMEOUT = TimeUnit.SECONDS.toMillis(5);
+
+    public TlsPromptHandler(@NonNull final PromptHandlerParameters parameters) {
+        super(new AadLoginComponentHandler(), parameters);
+        Logger.i(TAG, "Initializing Tls Prompt Handler..");
+    }
+
+    @Override
+    public void handlePrompt(@NonNull final String username, @NonNull final String password) {
+        final UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        final boolean loginHintProvided = !TextUtils.isEmpty(parameters.getLoginHint());
+        // handle browser
+        final BrowserChrome chrome = new BrowserChrome();
+        chrome.handleFirstRun();
+        // click on Open in chrome tabs.
+        UiAutomatorUtils.obtainUiObjectWithText("Microsoft");
+        try {
+            final UiObject menuButton = UiAutomatorUtils.obtainUiObjectWithResourceId("com.android.chrome:id/menu_button");
+            Thread.sleep(CHROME_MENU_BUTTON_RETRY_TIMEOUT);
+            menuButton.click();
+            Thread.sleep(CHROME_MENU_BUTTON_RETRY_TIMEOUT);
+            if (menuButton.exists()){
+                menuButton.click();
+            }
```

Comment:
I'd suggest putting this in a loop (rather than duplicating the code)

Reply:
948cdb496146cb7fe9ad30c038ac337726928722